app/grandchallenge/algorithms/templates/algorithms/result_list.html

```diff
@@ -129,11 +82,11 @@ <h2>Results for {{ algorithm.title }}</h2>
                         {% endif %}
                     </td>
                     <td>
-                        {% if job.result.images.all %}
-                            <ul class="list-unstyled">
-                                {% for image in job.result.images.all %}
+                        <ul class="list-unstyled">
+                            {% if result.images.all %}
+                                {% for image in result.images.all %}
```

Comment:
You probably want to use `{% for .. %}...{% empty %}` here, rather than wrap it in an `if..else` block. 

I think now `job.result.images.all` gets evaluated twice, which could lead to performance issues if there are many images.

Reply:
Oops!

Reply:
We should also do a prefetch_related on the queryset to stop that...